JSQMessagesViewController/Controllers/JSQMessagesViewController.m

```diff
@@ -357,7 +368,13 @@ - (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id<UIVi
 
 - (void)traitCollectionDidChange:(UITraitCollection *)previousTraitCollection {
     [super traitCollectionDidChange:previousTraitCollection];
-    [self jsq_resetLayoutAndCaches];
+    
+    // Only reset layout and caches if traits have actually changed;
+    // this method is called with a nil argument when the view is
+    // initially presented.
+    if (previousTraitCollection) {
+        [self jsq_resetLayoutAndCaches];
```

Comment:
Relayout is expensive, mostly because sizing is expensive and `jsq_resetLayoutAndCaches` blows away our message bubble size cache.  This cuts the presentation time for messages view almost in half.